.github/workflows/ci.yml

```diff
@@ -6,25 +6,30 @@ on:
       - main
   pull_request:
   schedule:
-    - cron: '0 12 * * *'
+    - cron: "0 12 * * *"
 
 jobs:
   test:
     strategy:
       matrix:
-        python:
-          - "3.7"
-          - "3.8"
-          - "3.9"
-          - "3.10"
-          - "3.11"
-    runs-on: ubuntu-latest
+        conf:
+          - { py: "3.7", os: "ubuntu-latest" }
+          - { py: "3.8", os: "ubuntu-latest" }
+          - { py: "3.9", os: "ubuntu-latest" }
+          - { py: "3.10", os: "ubuntu-latest" }
+          - { py: "3.11", os: "ubuntu-latest" }
+          # NOTE: We only test Windows and macOS on the latest Python;
+          # these primarily exist to ensure that we don't accidentally
+          # introduce Linux-isms into the development tooling.
+          - { py: "3.11", os: "windows-latest" }
+          - { py: "3.11", os: "macos-latest" }
+    runs-on: ${{ matrix.conf.os }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@83b7061638ee4956cf7545a6f7efe594e5ad0247 # v3.5.1
```

Comment:
IMO no need to pin to hashes here -- these are official actions, which makes them as close to "blessed" as possible.

Reply:
Yeah, `sigstore-python` uses a stricter policy. Let's drop down to a symbolic version here.